packages/zarathustra/skills/asylum_abci_app/behaviours.py

```diff
@@ -45,9 +48,28 @@
 
 
 TIMEZONE_UTC = UTC
+TELEGRAM_MSG_CHAR_LIMIT = 4_096
+MERMAID_DIAGRAMS = get_repo_root() / "specs" / "fsms" / "mermaid"
+BOUNTRY_REGEX_PATTERN = r"(?m)^\d PRIZE"
+SPONSOR_BOUNTY_DATA = get_repo_root() / "ethdenver-prizes.txt"
 
-MERMAID_DIAGRAMS = Path(__file__).parent / "mermaid_diagrams"
-THIS_MERMAID_PATH = MERMAID_DIAGRAMS / "asylum_abci_app.mmd"
+
+@functools.lru_cache
+def create_one_shot_examples(logger, n_examples: int = 10) -> str:
+    """Create one shot training examples of Mermaid diagrams for FSMs."""
+    example_data = []
+    for i, file in enumerate(islice(MERMAID_DIAGRAMS.glob("*"), n_examples)):
+        example_data.append(f"{i}. {file.stem}\n{file.read_text()}")
+    example_data = "\n\n".join(example_data)
+    logger.info(f"FSM Example Data: {example_data}")
+    return example_data
+
+
+@functools.lru_cache
+def get_bounty_info() -> list[str]:
+    """Get sponsor bounty data."""
+    content = SPONSOR_BOUNTY_DATA.read_text()
+    return re.split(BOUNTRY_REGEX_PATTERN, content)
```

Comment:
this needs to be refined, this is just raw text data with no proper pattern to split between bounties yet. Currently, in the behaviour where we create the prompt, I have hardcoded an index (`12`), which matches the first bounty of Story protocol